consul.yaml

```diff
@@ -20,32 +20,51 @@ pipeline:
       repository: https://github.com/hashicorp/consul
       tag: v${{package.version}}
       expected-commit: 5e08e229dbdaed5adf3ca99afe9df247c51507da
-      destination: consul
+      destination: ${{package.name}}
 
-  - runs: |
-      cd consul
-      # Mitigate GHSA-vvpx-j8f3-3w6h
-      go get golang.org/x/net@v0.7.0
-      # Mitigate GHSA-8cfg-vx93-jvxw
-      go get k8s.io/client-go@v0.20.1
+  - working-directory: ${{package.name}}
+    pipeline:
+      - runs: |
+          # Mitigate GHSA-vvpx-j8f3-3w6h
+          go get golang.org/x/net@v0.7.0
 
-      # Mitigate GHSA-ch7v-37xg-75ph
-      go get github.com/coredns/coredns@v1.10.1
-      go mod tidy
-      make linux
+          # Mitigate GHSA-8cfg-vx93-jvxw
+          go get k8s.io/client-go@v0.20.1
 
-  - runs: |
-      mkdir -p ${{targets.destdir}}/usr/bin
-      mv consul/pkg/bin/linux_*/consul ${{targets.destdir}}/usr/bin/consul
+          # Mitigate GHSA-ch7v-37xg-75ph
+          go get github.com/coredns/coredns@v1.10.1
 
-  - uses: strip
+          go mod tidy
+          make linux
+      - runs: |
+          mkdir -p ${{targets.destdir}}/usr/bin
+          mv ./pkg/bin/linux_*/consul ${{targets.destdir}}/usr/bin/consul
+      - uses: strip
 
 subpackages:
-  - name: consul-oci-entrypoint
+  - name: ${{package.name}}-oci-entrypoint
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/usr/bin
-          mv consul/.release/docker/docker-entrypoint.sh ${{targets.subpkgdir}}/usr/bin/
+          cp consul/.release/docker/docker-entrypoint.sh "${{targets.subpkgdir}}/usr/bin/"
+    dependencies:
+      runtime:
+        - busybox
+        - dumb-init
+        - su-exec
+
+  - name: ${{package.name}}-compat
+    description: "Compatiblity package for locating binaries according to upstream helm charts"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}/usr/local/bin"
+          cp "${{targets.destdir}}/usr/bin/consul" "${{targets.subpkgdir}}/usr/local/bin"
```

Comment:
Should this be a symlink instead of copying the binary?

Reply:
Yep good idea

Reply:
Actually, I sorta take it back. You're right, it should prefer a symlink. However the entrypoint is expecting `/bin/consul` and it can't be a symlink because the entrypoint sets capabilities on it. On direction from @kaniini I just put the binary in `/bin` in the main package